tracer/src/Datadog.Tracer.Native/iast/hardcoded_secrets_method_analyzer.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+#include "../../../../shared/src/native-src/pal.h"
+#include "method_analyzer.h"
+#include <atomic>
+#include <map>
+#include <set>
+#include "method_analyzer.h"
+#include "iast_util.h"
+
+using namespace shared;
+
+namespace iast
+{
+    struct UserString
+    {
+        shared::WSTRING location;
+        shared::WSTRING value;
+    };
+
+    struct UserStringInterop
+    {
+        const WCHAR* location;
+        const WCHAR* value;
+    };
+
+    class HardcodedSecretsMethodAnalyzer : public MethodAnalyzer
+    {
+    private:
+        CS _cs;
+        std::vector<UserString> _userStrings;
+        std::vector<UserString> _deliveredUserStrings;
+    public:
+        static HardcodedSecretsMethodAnalyzer* Instance;
+
+        HardcodedSecretsMethodAnalyzer();
+        virtual ~HardcodedSecretsMethodAnalyzer();
```

Comment:
nit you can remove the `virtual` and add `override`

Reply:
You are right, but I've never done it in virtual destructors. In fact, there's quite a lot of discussion about that ([here's an example](https://github.com/isocpp/CppCoreGuidelines/issues/721))